pkg/auth/oauth/handlers/default_auth_handler.go

```diff
@@ -32,6 +33,22 @@ func NewUnionAuthenticationHandler(passedChallengers map[string]AuthenticationCh
 	return &unionAuthenticationHandler{challengers, redirectors, errorHandler}
 }
 
+const (
+	// WarningHeaderMiscCode is the code for "Miscellaneous warning", which may be displayed to human users
+	WarningHeaderMiscCode = "199"
+	// The name of the agent adding the warning header
+	WarningHeaderOpenShiftSource = "OpenShift"
+
+	warningHeaderCodeIndex  = 1
+	warningHeaderAgentIndex = 2
+	warningHeaderTextIndex  = 3
+)
+
+var (
+	// http://tools.ietf.org/html/rfc2616#section-14.46
+	warningRegex = regexp.MustCompile(`([0-9]{3}) ([^ ]+) "([^"]+)".*`)
```

Comment:
Oh great, it's used.  Three numbers, a space, one or more non-space characters, a space, a non-empty quoted string, and then whatever I want?  If you're going to do this, handle the date instead of punting it and have fixed start and end.

Surely this is built into some library.